c7n/resources/s3.py

```diff
@@ -1922,9 +1922,19 @@ def get_event_buckets(self, client, trails):
 
         For wildcard trails the bucket name is ''.
         """
+        regions = {t.get('HomeRegion') for t in trails}
+        clients = {None: client}
+        for region in regions:
+            if region == client._client_config.region_name:
+                clients[region] = client
+            else:
+                clients[region] = local_session(
```

Comment:
probably not meaningful, but its actually a bit faster, to create the session once, and then create the client for each region using the same session.

Reply:
I fixed it to re-use the session.  Is the client cheap enough to create that I don't need to cache them?

Reply:
client is cheap, its an amortized cost of loading the data models from the json files and creating the class structure, its memoized on the session.